### DISCUSSION TITLE
Keycloak 2FA SMS Authenticator

### DISCUSSION THREAD

**Discussion by bmunishwar** [keycloak/keycloak#20859] [Q&A]
How to implement 2nd-factor authentication with a OTP/code/token send via SMS on mobile number or email ID ?

**Comment by danielFesenmeyer** [comment]
By default, Keycloak does not provide such authentication.

But the following repo looks promising: https://github.com/netzbegruenung/keycloak-mfa-plugins. It seems they provide frequent releases. Btw, the code is based on a demo implemented by @dasniko: https://github.com/dasniko/keycloak-2fa-sms-authenticator

**Reply by Nefcanto** [reply]
I wish Keycloak would implement this by default. Though less secure, it's a very common mechanism for login. Even a lot of services use phone-OTP for login.

**Reply by roamerz** [reply]
I’m not sure why they don’t. It’s not a big lift compared to some things and although not the best option for security it would be another option for those who for whatever reason didn’t want to use a TOPT.

**Reply by svenseeberg** [reply]
Feel free to create issues and/or PRs in https://github.com/netzbegruenung/keycloak-mfa-plugins. We plan to maintain the SMS plugin for the foreseeable future.

**Comment by roamerz** [comment]
If anyone is interested, I messed around a bit with @dasniko code. Still a work in progress and use at your own risk.
https://github.com/roamerz/Keycloak-SMS-Authenticator

**Comment by codespearhead** [comment]
Can you please close this issue/discussion [1][2]?

This question has been asked several times before. You can find previous discussions by searching for “sms” or “email” in the GitHub search bar of the Keycloak repository [3][4].

Currently, there is no official support for using email or phone numbers for authentication in Keycloak. Therefore, any issues you encounter with third-party solutions should be reported in their respective repositories, not here.

If you're interested in this feature, feel free to share your use case here [5], and upvote it by adding a ⬆️ to the discussion’s description after reading this answer carefully [6].

***

[1] https://docs.github.com/en/issues/tracking-your-work-with-issues/administering-issues/closing-an-issue
[2] https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions#closing-a-discussion
[3] https://github.com/search?q=repo%3Akeycloak%2Fkeycloak+%22SMS%22&type=issues
[4] https://github.com/search?q=repo%3Akeycloak%2Fkeycloak+%22SMS%22&type=discussions
[5] https://github.com/keycloak/keycloak/discussions/27152
[6] https://github.com/keycloak/keycloak/discussions/27152#discussioncomment-10923758
